src/hook/useInput.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+type ValidatorFn = (value: string) => string | null;
+
+function useInput(validator: ValidatorFn, initialValue ='') {
+  const [value, setValue] = useState(initialValue);
+  const [error, setError] = useState<string | null>(' ');
+  const [touched, setTouched] = useState(false);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+    const errorMsg = validator(e.target.value);
+    setError(errorMsg);
+  };
+
+  const onBlur = () => {
+    if(!touched) setTouched(true);
+    const errorMsg = validator(value);
+    setError(errorMsg);
+  };
+
+  const isValid = touched ? !error : true
```

Comment:
```suggestion
  const isValid = touched || !error
```

요로코롬 표현하면 될 것 같네요..!?

Reply:
이부분은 저도 처음에 저런 방식으로 했었는데요.
저는 isValid를 에러 메세지의 표현 유무 뿐만 아니라 에러가 있을때 빨간 줄로 표시하는 UI 에도 사용을 했습니다.
(에러 메세지는 의도대로 동작하지만) 처음 로그인 화면 들어갈때부터 빨간줄이 떠있는 상태가 됩니다!